appservice/multiplexer.py

```diff
@@ -94,6 +96,18 @@ async def handle_session_status(request):
         return PlainTextResponse('unknown session ID', status_code=404)
 
 
+@app.route(f'{config.ROUTE_API}/sessions/{{sessionid}}/wait-status')
+async def handle_session_wait_status(request):
+    sessionid = request.path_params['sessionid']
+    if sessionid not in SESSIONS:
+        return PlainTextResponse('unknown session ID', status_code=404)
+
+    f = asyncio.get_running_loop().create_future()
+    WAIT_FUTURES.setdefault(sessionid, []).append((SESSIONS[sessionid], f))
```

Comment:
Setting old_state that way is racy and bogus. We need something reliable where the caller says "my last information was state X, tell me when it becomes something else", i.e. this needs to be passed as GET parameter.